docker/build/devdeps.manylinux.Dockerfile

```diff
@@ -44,15 +44,19 @@ RUN git clone --filter=tree:0 https://github.com/llvm/llvm-project /llvm-project
 # a wrapper script so that the path that we set CC and CXX to is independent 
 # on the installed toolchain. Unfortunately, a symbolic link won't work.
 # Using update-alternatives for c++ and cc could maybe be a better option.
-ENV LLVM_INSTALL_PREFIX=/opt/llvm
-RUN if [ "$toolchain" == 'gcc11' ]; then \
-        dev_tools=gcc-toolset-11 && CC=$(which gcc | sed 's/[0-9]\{1,2\}/11/g') && CXX=$(which g++ | sed 's/[0-9]\{1,2\}/11/g'); \
-    elif [ "$toolchain" == 'clang15' ]; then \
-        dev_tools=clang && CC=$(which clang-15) && CXX=$(which clang++-15); \
+ENV LLVM_INSTALL_PREFIX=/usr/local/llvm
+RUN if [ "${toolchain#gcc}" != "$toolchain" ]; then \
+        gcc_version=`echo $toolchain | grep -o '[0-9]*'` && \
+        if [ -z "$(which gcc 2> /dev/null | grep $gcc_version)" ]; then \
+            dnf install -y --nobest --setopt=install_weak_deps=False gcc-toolset-$gcc_version && \
+            enable_script=`find / -path '*gcc*' -path '*'$gcc_version'*' -name enable` && . "$enable_script"; \
+        fi && \
+        CC="$(which gcc)" && CXX="$(which g++)"; \
+    elif [ "$toolchain" == 'clang16' ]; then \
+        dnf install -y --nobest --setopt=install_weak_deps=False clang-16.0.6 && \
+        CC="$(which clang-16)" && CXX="$(which clang++-16)"; \
```

Comment:
Should we be hardcoding LLVM 16 into this dockerfile script?

Reply:
I am not sure I understand. This hardcodes nothing. The dockerfile takes a toolchain argument so it could in principle be built with different compilers (though maybe we should just remove that for this one), and this line here merely contains the logic to install the requested toolchain. If anything, we should remove the option to choose. This has nothing to do with the LLVM version that the CUDA-Q build depends on; this is the compiler to build CUDA-Q.

Reply:
I see a `-16` hardcoded in this command.

Some day, it would be nice to not be nailed to clang-16, but the more places we hardcode `-16` in files, the more painful it will be to upgrade to a modern llvm-project.

Reply:
The -16 in `which clang-16` or in the `dnf install` is inside a code block that executes conditionally on clang16 having been explicitly requested as the toolchain. Once we update to a newer version, we will add another elif clause for that clang version. There isn't a more generic way of installing clang here, since installations of different version is a bit non-uniform (e.g. install clang-16 is not sufficient here if I remember correctly).